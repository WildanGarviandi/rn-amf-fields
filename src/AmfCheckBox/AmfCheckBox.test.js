import React from 'react'
import AmfCheckBox from './AmfCheckBox'
import renderer from 'react-test-renderer'

const value = ['foo']
const label = 'Label'
const items = [{
  label: 'foo',
  value: 'foo'
}, {
  label: 'bar',
  value: 'bar'
}]
const onPress = jest.fn()

test('AmfCheckBox simple render', () => {
  let props = { value, label, items, onPress }
  props.options = [
    { label: 'Label1', value: 'value2' },
    { label: 'Label2', value: 'value3' },
  ]
  
  const component = renderer.create(<AmfCheckBox {...props} />)
  
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test.skip('AmfCheckBox with validation', () => {
  let props = { label, items, onPress,
    required: true,
    value: []
  }
  
  const component = renderer.create(<AmfCheckBox {...props} />)
  
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
  
/**
   *
   */
test('AmfCheckBox without options', () => {
  let props = { value, label, items, onPress }
  
  console.error = jest.fn()
  
  expect(() => {
    const component = renderer.create(<AmfCheckBox {...props} />)
    let tree = component.toJSON()
  }).toThrowErrorMatchingSnapshot()
})
