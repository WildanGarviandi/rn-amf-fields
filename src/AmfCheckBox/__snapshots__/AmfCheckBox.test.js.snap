// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfCheckBox simple render 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
        "marginRight": 16,
      }
    }
  >
    Label
  </Text>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "padding": 5,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
          }
        }
      >
        Label1
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
          }
        }
      >
        Label2
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  />
</View>
`;

exports[`AmfCheckBox without options 1`] = `"Field \\"Label\\" (AmfCheckBox) doesnt have options"`;
